### PR TITLE
WIP: Aerospike auth support, latency and throughput collection

### PR DESCRIPTION
**Pull request by usdin**
### Required for all PRs:

- [ ] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] README.md updated (if adding a new plugin)

### CODE REVIEW

File: plugins/inputs/aerospike/aerospike.go [hunk 1]
```diff
@@ -54,7 +57,11 @@ func (a *Aerospike) Gather(acc telegraf.Accumulator) error {
 	return errChan.Error()
 }
 
-func (a *Aerospike) gatherServer(hostport string, acc telegraf.Accumulator) error {
+func (a *Aerospike) gatherServer(rawurl string, acc telegraf.Accumulator) error {
+	rawurl = "aero://" + rawurl
+	urlInfo, err := url.Parse(rawurl)
```
Comment (sparrc) [inline plugins/inputs/aerospike/aerospike.go]: you should return this error if it's not nil

File: plugins/inputs/aerospike/aerospike.go [hunk 2]
```diff
@@ -24,7 +27,7 @@ var sampleConfig = `
   ## Aerospike servers to connect to (with port)
   ## This plugin will query all namespaces the aerospike
   ## server has configured and get stats for them.
-  servers = ["localhost:3000"]
+  servers = ["user:password@localhost:3000","localhost:3000"]
```
Comment (sparrc) [inline plugins/inputs/aerospike/aerospike.go]: don't add another server to the list, just document in a comment that this is how to format user/pass combos

Comment (sparrc) [inline plugins/inputs/aerospike/aerospike.go]: add `aero://` prefix in sample config

File: plugins/inputs/aerospike/aerospike.go [hunk 3]
```diff
@@ -54,7 +57,11 @@ func (a *Aerospike) Gather(acc telegraf.Accumulator) error {
 	return errChan.Error()
 }
 
-func (a *Aerospike) gatherServer(hostport string, acc telegraf.Accumulator) error {
+func (a *Aerospike) gatherServer(rawurl string, acc telegraf.Accumulator) error {
+	rawurl = "aero://" + rawurl
```
Comment (sparrc) [inline plugins/inputs/aerospike/aerospike.go]: check if it already has a prefix before adding one

File: plugins/inputs/aerospike/aerospike.go [hunk 4]
```diff
@@ -65,67 +72,231 @@ func (a *Aerospike) gatherServer(hostport string, acc telegraf.Accumulator) erro
 		iport = 3000
 	}
 
-	c, err := as.NewClient(host, iport)
+	var c *as.Client
+	if urlInfo.User != nil {
+		userInfo := urlInfo.User
+		clientPolicy := as.NewClientPolicy()
+		clientPolicy.User = userInfo.Username()
+		clientPolicy.Password, _ = userInfo.Password()
```
Comment (sparrc) [inline plugins/inputs/aerospike/aerospike.go]: could you add TLS support here as well? see https://github.com/influxdata/telegraf/blob/master/internal/internal.go#L118 and see the influxdb output plugin sample config for how to format the config options.

Comment (usdin) [inline plugins/inputs/aerospike/aerospike.go]: I have a question, Influxdb output plugin have only one cluster so one tls config works. Aerospike plugin supports multiple cluster. Should I modify it to take nodes of one cluster only.

Comment (sparrc) [inline plugins/inputs/aerospike/aerospike.go]: no, just use the tls config for all servers specified in one plugin. If users need multiple tls configs they can specify multiple [[inputs.aerospike]] instances

**Comment by deric** [comment]
@usdin any progress on this? Do you need some help with the requested changes?

**Comment by jeremiegoldberg** [comment]
Hi, any news on this plugin ?

Thank you !

**Comment by sjwang90** [comment]
Closing due to lack of activity and discussion.  Feel free to reopen if desired.

**Pull request closed without merging**